crates/bevy_ui/src/render/mod.rs

```diff
@@ -344,8 +344,9 @@ pub fn prepare_uinodes(
             positions[3] + positions_diff[3].extend(0.),
         ];
 
-        let rect_size_v4 = uinode_rect.size().extend(0.).extend(0.);
-        let transformed_rect_size = (extracted_uinode.transform * rect_size_v4).xy().extend(1.);
+        
+        let transformed_rect_size =
+            (extracted_uinode.transform * uinode_rect.size().extend(0.).extend(0.)).xy();
```

Comment:
```suggestion
        let transformed_rect_size = extracted_uinode.transform.xy() * uinode_rect.size();
```
I didn't check if that compiles, but instead of doing the multiplication in vec4 then going back to a vec2, wouldn't it work to go straight to a vec2?

Reply:
Or alternatively we could just use the x and y fields directly perhaps?